prefect_twitter/media.py

```diff
@@ -0,0 +1,71 @@
+"""This is a module for interacting with Twitter media"""
+
+from functools import partial
+from pathlib import Path
+from typing import TYPE_CHECKING, Union
+
+from anyio import to_thread
+from prefect import get_run_logger, task
+
+if TYPE_CHECKING:
+    from io import IOBase
+
+    from prefect_twitter import TwitterCredentials
+
+
+@task
+async def media_upload(
+    filename: Union[Path, str],
+    twitter_credentials: "TwitterCredentials",
+    file: "IOBase" = None,
+    chunked: bool = None,
+    **kwargs: dict
+) -> int:
+    """
+    Use this to upload media to Twitter. Chunked media upload
+    is automatically used for videos.
+
+    Args:
+        filename: The filename of the image to upload.
+            This will automatically be opened unless file is specified.
+        twitter_credentials: Credentials to use for authentication with Tweepy.
+        file: A file object, which will be used instead of opening filename.
+            Note, filename is still required, for MIME type detection and
+            to use as a form field in the POST data.
+        chunked: Whether or not to use chunked media upload.
+            Videos use chunked upload regardless of this parameter.
+        kwargs: Additional keyword arguments to pass to media_upload.
+    Returns:
+        The media ID.
+
+    Example:
+        Uploads an image from a file path to Twitter.
+        ```python
+        from prefect import flow
+        from prefect_twitter import TwitterCredentials
+        from prefect_twitter.media import media_upload
+
+        @flow
+        def example_media_upload_flow():
+            twitter_credentials = TwitterCredentials(
+                consumer_key="consumer_key",
+                consumer_secret="consumer_secret",
+                access_token="access_token",
+                access_token_secret="access_token_secret"
+            )
+            media_id = media_upload("/path/to/prefection.jpg", twitter_credentials)
+            return media_id
+
+        example_update_status_flow()
+        ```
+    """
+    logger = get_run_logger()
+    logger.info("Uploading media named %s", filename)
+
+    api = twitter_credentials.get_api()
+    partial_media = partial(
+        api.media_upload, filename=filename, file=file, chunked=chunked, **kwargs
+    )
+    media = await to_thread.run_sync(partial_media)
+    media_id = media.media_id
+    return media_id
```

Comment:
I think since we don't have a task to retrieve media using an ID, we should return the whole object.

Reply:
From my perspective, passing the media_id down to the update_status is really unfriendly to the user (additional of two confusing awaits and `result()`) so I updated it from object to the ID.
```
Completed(message=None, type=COMPLETED, result=Media(_api=<tweepy.api.API object at 0x12a485a60>, media_id=1507424499620265988, media_id_string='1507424499620265988', size=59818, expires_after_secs=86400, image={'image_type': 'image[/png]()', 'w': 6774, 'h': 1731}), task_run_id=83d245ad-9e2c-4713-96fa-7d985cc549d9)
```
I think the main goal of upload media is to just pass to `update_status`; I don't think there's other purposes.

Reply:
I guess I can implement https://docs.tweepy.org/en/stable/api.html#tweepy.API.get_media_upload_status in another PR?

Reply:
Can you create an issue to track adding that additional task?